app/ante/eth.go

```diff
@@ -213,8 +218,11 @@ func (egcd EthGasConsumeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simula
 		gasPool.ConsumeGas(ctx.GasMeter().GasConsumedToLimit(), "gas pool check")
 	}
 
+	// Set newCtx.GasMeter to 0, with a limit of GasWanted (gasLimit)
+	newCtx = ctx.WithGasMeter(sdk.NewGasMeter(gasTx.GetGas()))
```

Comment:
I'd add the sum the gas values in the form loop above as there could be a potential case where a multi msg tx is crafted in a way that the tx gas is less than the sum of the msgs gas limits

Reply:
Done.